packages/nodejs/src/__tests__/config.test.ts

```diff
@@ -27,124 +54,211 @@ describe("Configuration", () => {
-      filterParameters: [],
-      filterSessionData: [],
-      ignoreActions: [],
-      ignoreErrors: [],
-      ignoreNamespaces: [],
-      log: "file",
-      logPath: "/tmp",
-      transactionDebugMode: false
-    }
-
-    const expectedInitialConfig = {
-      name,
-      apiKey
-    }
-
-    const expectedConfig = {
-      ...expectedDefaultConfig,
-      ...expectedInitialConfig
-    }
-
-    it("loads all default options when no options are overwritten", () => {
-      expect(config.data).toEqual(expectedConfig)
+  describe("with initial config options", () => {
+    it("loads initial config otpions", () => {
+      const initialOptions = {
+        debug: true,
+        enableStatsd: true
+      }
+      const options = {
+        ...expectedDefaultConfig,
+        ...initialOptions
+      }
+      config = new Configuration({
+        ...initialOptions
+      })
+
+      expect(config.data).toEqual(options)
 
       expect(config.sources.default).toEqual(expectedDefaultConfig)
-      expect(config.sources.initial).toEqual(expectedInitialConfig)
+      expect(config.sources.initial).toEqual(initialOptions)
       expect(config.sources.env).toEqual({})
     })
+  })
 
-    describe("overwrites default values", () => {
-      const overwrittenConfig = {
+  describe("with config in the environment", () => {
+    it("loads configuration from the environment", () => {
+      process.env["APPSIGNAL_DEBUG"] = "true"
+      process.env["APPSIGNAL_ENABLE_STATSD"] = "true"
+      const envOptions = {
         debug: true,
         enableStatsd: true
       }
-
       const expectedConfig = {
         ...expectedDefaultConfig,
-        ...overwrittenConfig
+        ...envOptions
       }
+      config = new Configuration({})
 
-      it("with initial values", () => {
-        config = new Configuration({
-          ...overwrittenConfig
-        })
+      expect(config.data).toEqual(expectedConfig)
+
+      expect(config.sources.default).toEqual(expectedDefaultConfig)
+      expect(config.sources.initial).toEqual({})
+      expect(config.sources.env).toEqual(envOptions)
+    })
+  })
+
+  describe("logFilePath", () => {
+    it("uses the default log file path", () => {
+      new Configuration({ name, apiKey })
 
-        expect(config.data).toEqual(expectedConfig)
+      expect(env("_APPSIGNAL_LOG_FILE_PATH")).toEqual("/tmp/appsignal.log")
+    })
 
-        expect(config.sources.default).toEqual(expectedDefaultConfig)
-        expect(config.sources.initial).toEqual(overwrittenConfig)
-        expect(config.sources.env).toEqual({})
+    describe("with logPath option", () => {
+      beforeEach(() => {
+        new Configuration({ logPath: "/other_path" })
       })
 
-      it("with environment values", () => {
-        process.env["APPSIGNAL_DEBUG"] = "true"
-        process.env["APPSIGNAL_ENABLE_STATSD"] = "true"
+      it("uses the overwritten path", () => {
+        expect(env("_APPSIGNAL_LOG_FILE_PATH")).toEqual(
```

Comment:
Nitpick: these expectations could be done against the config object's `.logFilePath` method, instead of fetching the value from the environment (there's tests checking that it is being fetched from the environment later in the file)